File: server-scala-play-server/app/api/StopApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.BigDecimal
import model.Stop

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@Singleton
class StopApiController @Inject()(cc: ControllerComponents, api: StopApi) extends AbstractController(cc) {
  /**
    * GET /api/:version/stop/:id
    * @param id the id of the stop to get
    */
  def getStop(version: BigDecimal, id: Long): Action[AnyContent] = Action { request =>
    def executeApi(): Stop = {
      api.getStop(version, id)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * PUT /api/:version/stop/:id
    * @param id the id of the stop to update
    */
  def updateStop(version: BigDecimal, id: Long): Action[AnyContent] = Action { request =>
    def executeApi(): Stop = {
      val body = request.body.asJson.map(_.as[Stop])
      api.updateStop(version, id, body)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}

File: server-java-play-framework/app/controllers/StopApiController.java
package controllers;

import apimodels.Stop;

import com.typesafe.config.Config;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import play.libs.Files.TemporaryFile;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import javax.validation.Valid;
import com.typesafe.config.Config;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-11T19:25:37.664944661Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
public class StopApiController extends Controller {
    private final StopApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Config configuration;

    @Inject
    private StopApiController(Config configuration, StopApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }

    @ApiAction
    public Result getStop(Http.Request request, Long id) throws Exception {
        return imp.getStopHttp(request, id);
    }

    @ApiAction
    public Result updateStop(Http.Request request, Long id) throws Exception {
        JsonNode nodebody = request.body().asJson();
        Stop body;
        if (nodebody != null) {
            body = mapper.readValue(nodebody.toString(), Stop.class);
            if (configuration.getBoolean("useInputBeanValidation")) {
                OpenAPIUtils.validate(body);
            }
        } else {
            body = null;
        }
        return imp.updateStopHttp(request, id, body);
    }

}
